### PR TITLE
added a way to set logging information synced to remote host and more…

### PR DESCRIPTION
**Pull request by heejaechang**
… logging for error list.

<details>

### Customer scenario

This change is not for user experience but adding more logging and a way to set logging options on remote host.

### Bugs this fixes

N/A

### Workarounds, if any

N/A

### Risk

this won't affect normal users using VS. it requires debugging vsix to be installed to enable these options.

### Performance impact

N/A

### Is this a regression from a previous update?

No

### Root cause analysis

this is not a bug fix. I am adding this to make error list diagnostics easier.

### How was the bug found?

there is no bug for this. 
</details>

### CODE REVIEW

**Comment by heejaechang** [comment]
can you take a look?

**Comment by heejaechang** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by heejaechang** [comment]
@dotnet/roslyn-infrastructure test failure
https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_vs-integration_prtest/9360/

Roslyn.VisualStudio.IntegrationTests.Basic.BasicEncapsulateField.EncapsulateThroughLightbulbDefinitionsOnly

**Comment by heejaechang** [comment]
retest windows_debug_vs-integration_prtest  please

**Comment by heejaechang** [comment]
@jinujoseph this doesn't affect testing. but want to be in so that when things repro again, I can investigate easier.

**Comment by heejaechang** [comment]
ping? basically more logging added to code and add ability to enable/disable logging in both VS and remote host at the same time.

**Comment by jinujoseph** [comment]
@Pilchie for ask mode approval

File: src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseDiagnosticListTable.LiveTableDataSource.cs [hunk 1]
```diff
@@ -537,6 +541,17 @@ public bool TryCreateStringContent(int index, string columnName, bool singleColu
...
+
+            private static string GetDiagnosticUpdatedMessage(DiagnosticsUpdatedArgs e)
+            {
+                var id = e.Id.ToString();
+                if (e.Id is AnalyzerUpdateArgsId analyzer)
+                {
+                    id = analyzer.Analyzer.ToString();
+                }
+
+                return $"{e.Workspace.Kind} {id} {e.Kind} {(object)e.DocumentId ?? e.ProjectId} {e.Diagnostics.Length}";
```
Comment (Pilchie) [inline src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseDiagnosticListTable.LiveTableDataSource.cs]: How often does this get called - do the allocations here matter?  @sharwell?

Comment (heejaechang) [reply src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseDiagnosticListTable.LiveTableDataSource.cs]: without option on, this never get called. it uses our Logger framework, which only build string to log when logging option is on.

**Comment by heejaechang** [comment]
I would like this to be in so that if this repro again (https://devdiv.visualstudio.com/DevDiv/_workitems/edit/522869) I can investigate more easily.

**Comment by heejaechang** [comment]
@Pilchie do I need to go through ask mode for this? or it is okay without it since it is not UX affecting code.

File: src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseDiagnosticListTable.LiveTableDataSource.cs (hunk 1 above)
Comment (Pilchie) [reply src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseDiagnosticListTable.LiveTableDataSource.cs]: Got it, thanks.

**Comment by Pilchie** [comment]
Approved.

**Pull request merged**